miio/miioprotocol.py

```diff
@@ -90,18 +90,18 @@ def send_handshake(self, *, retry_count=3) -> Message:
         return m
 
     @staticmethod
-    def discover(addr: str = None, timeout: int = 5) -> Any:
+    def discover(addr: str = None, is_broadcast: bool = True, timeout: int = 5) -> Any:
         """Scan for devices in the network. This method is used to discover supported
         devices by sending a handshake message to the broadcast address on port 54321.
         If the target IP address is given, the handshake will be send as an unicast
         packet.
 
         :param str addr: Target IP address
         """
-        is_broadcast = addr is None
         seen_addrs = []  # type: List[str]
         if is_broadcast:
-            addr = "<broadcast>"
+            if not addr:
+                addr = "<broadcast>"
```

Comment:
I don't really like having nested ifs but I think it's fine until this method gets cleaned up... At the moment it is trying to perform both discovery & handshaking.

Reply:
Agree, might be nice to split it up, but this seemed like the most minimal change